core/ibft.go

```diff
@@ -432,35 +439,45 @@ func (i *IBFT) waitForRCC(
 // and constructs a RCC if possible
 func (i *IBFT) handleRoundChangeMessage(view *proto.View) *proto.RoundChangeCertificate {
 	var (
-		height = view.Height
-		round  = view.Round
+		height              = view.Height
+		hasAcceptedProposal = i.state.getProposal() != nil
 	)
 
-	isValidFn := func(msg *proto.Message) bool {
+	isValidMsgFn := func(msg *proto.Message) bool {
 		proposal := messages.ExtractLastPreparedProposedBlock(msg)
 		certificate := messages.ExtractLatestPC(msg)
 
 		// Check if the prepared certificate is valid
-		if !i.validPC(certificate, round, height) {
+		if !i.validPC(certificate, msg.View.Round, height) {
 			return false
 		}
 
 		// Make sure the certificate matches the proposal
 		return i.proposalMatchesCertificate(proposal, certificate)
 	}
 
-	msgs := i.messages.GetValidMessages(
-		view,
-		proto.MessageType_ROUND_CHANGE,
-		isValidFn,
+	isValidRCCFn := func(round uint64, msgs []*proto.Message) bool {
+		// In case of that ROUND-CHANGE message's round match validator's round
+		// Accept such messages only if the validator has not accepted a proposal at the round
+		if round == view.Round && hasAcceptedProposal {
```

Comment:
I suppose this is according to algorithm, but just to check:  If a node has received a proposal it will disregard any kind of Round Change messages?  Is it possible that the disregarded messages create quorum and the node should proceed to the new round?

Reply:
If node has received a valid PROPOSAL message for the validator's current round, the validator ignore ROUND-CHANGE messages for the same round. But still the validator can move to new round which is bigger than the current round once the validator received enough ROUND-CHANGE messages meeting quorum